tests/unit/array/storage/elastic/test_del.py

```diff
@@ -0,0 +1,96 @@
+import docarray.array.storage.elastic.backend as elastic_backend
+from docarray.array.storage.base.helper import Offset2ID
+
+MOCK_ES_OFFSET_INDEX = {
+    0: {'_id': 0, '_source': {'blob': 'r0'}},
+    1: {'_id': 1, '_source': {'blob': 'r1'}},
+    2: {'_id': 2, '_source': {'blob': 'r2'}},
+    3: {'_id': 3, '_source': {'blob': 'r3'}},
+    4: {'_id': 4, '_source': {'blob': 'r4'}},
+    5: {'_id': 5, '_source': {'blob': 'r5'}},
+    6: {'_id': 6, '_source': {'blob': 'r6'}},
+    7: {'_id': 7, '_source': {'blob': 'r7'}},
+    8: {'_id': 8, '_source': {'blob': 'r8'}},
+}
+
+BUG_OFFSET_INDEX_AFTER_DELETION = {
+    0: {'_id': 0, '_source': {'blob': 'r3'}},
+    1: {'_id': 1, '_source': {'blob': 'r4'}},
+    2: {'_id': 2, '_source': {'blob': 'r5'}},
+    3: {'_id': 3, '_source': {'blob': 'r6'}},
+    4: {'_id': 4, '_source': {'blob': 'r7'}},
+    5: {'_id': 5, '_source': {'blob': 'r8'}},
+    6: {'_id': 6, '_source': {'blob': 'r6'}},
+    7: {'_id': 7, '_source': {'blob': 'r7'}},
+    8: {'_id': 8, '_source': {'blob': 'r8'}},
+}
+
+EXPECTED_INDEX_AFTER_DELETION = {
+    0: {'_id': 0, '_source': {'blob': 'r3'}},
+    1: {'_id': 1, '_source': {'blob': 'r4'}},
+    2: {'_id': 2, '_source': {'blob': 'r5'}},
+    3: {'_id': 3, '_source': {'blob': 'r6'}},
+    4: {'_id': 4, '_source': {'blob': 'r7'}},
+    5: {'_id': 5, '_source': {'blob': 'r8'}},
+}
+
+
+class MockElasticIndices:
+    def __init__(self, **kwargs):
+        pass
+
+    def exists(self, **kwargs):
+        return True
+
+    def refresh(self, **kwargs):
+        pass
+
+
+class MockElasticClient:
+    bulk_call_count = 0
+
+    def __init__(self, **kwargs):
+        self.indices = MockElasticIndices()
+
+    def count(self, **kwargs):
+        return {'count': len(MOCK_ES_OFFSET_INDEX)}
+
+    @classmethod
+    def bulk(cls, *args, **kwargs):
+        cls.bulk_call_count += 1
+
+        requests = args[1]
+
+        for request in requests:
+            if request['_op_type'] == 'index':
+                MOCK_ES_OFFSET_INDEX[request['_id']] = {
+                    '_id': request['_id'],
+                    '_source': {'blob': request['blob']},
+                }
+            elif request['_op_type'] == 'delete':
+                del MOCK_ES_OFFSET_INDEX[request['_id']]
+
+        if cls.bulk_call_count == 1:  # 1st bulk api call, update offset index
+            assert MOCK_ES_OFFSET_INDEX == BUG_OFFSET_INDEX_AFTER_DELETION
```

Comment:
I think this assert is not important at all and only provides noise

Reply:
noted